accumulo-store/src/main/java/gaffer/accumulostore/key/core/impl/byteEntity/ByteEntityAccumuloElementConverter.java

```diff
@@ -159,13 +159,27 @@ protected boolean getSourceAndDestinationFromRowKey(final byte[] rowKey, final b
         } catch (final NumberFormatException e) {
             throw new AccumuloElementConversionException("Error parsing direction flag from row key - " + e);
         }
-        if (directionFlag == ByteEntityPositions.UNDIRECTED_EDGE) {
-            // Edge is undirected
+        if (directionFlag == ByteEntityPositions.CORRECT_WAY_UNDIRECTED_EDGE) {
+            // Edge is undirected and the first identifier is the source of the edge
             sourceValueDestinationValue[0] = ByteArrayEscapeUtils
                     .unEscape(Arrays.copyOfRange(rowKey, 0, positionsOfDelimiters[0]));
             sourceValueDestinationValue[1] = ByteArrayEscapeUtils
                     .unEscape(Arrays.copyOfRange(rowKey, positionsOfDelimiters[1] + 1, positionsOfDelimiters[2]));
             return false;
+        } else if (directionFlag == ByteEntityPositions.INCORRECT_WAY_UNDIRECTED_EDGE) {
+            // Edge is undirected and the second identifier is the source of the edge
+            int src = 1;
+            int dst = 0;
+            if (options != null && options.containsKey(AccumuloStoreConstants.OPERATION_RETURN_MATCHED_SEEDS_AS_EDGE_SOURCE)
```

Comment:
This is becoming less readable.  Can we split out some of these calls into parameters which can be used in the conditional?